homeassistant/components/sleepiq/number.py

```diff
@@ -138,6 +162,18 @@ def _get_foot_warming_unique_id(bed: SleepIQBed, foot_warmer: SleepIQFootWarmer)
         get_name_fn=_get_foot_warming_name,
         get_unique_id_fn=_get_foot_warming_unique_id,
     ),
+    CORE_CLIMATE_TIMER: SleepIQNumberEntityDescription(
+        key=CORE_CLIMATE_TIMER,
+        native_min_value=30,
+        native_max_value=SleepIQCoreClimate.max_core_climate_time,
```

Comment:
It shouldn't really be set here since then it won't use the value from the specific object.  Probably set it at line 224 and use `core_climate.max_core_climate_time`

Reply:
The dataclass for SleepIQNumberEntityDescription is frozen. 

When attempting to reset the value I get: 

```
2024-01-28 21:38:02.230 ERROR (MainThread) [homeassistant.components.number] Error while setting up sleepiq platform for number
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 326, in _async_setup_platform
    await asyncio.shield(task)
  File "/workspaces/core/homeassistant/components/sleepiq/number.py", line 225, in async_setup_entry
    entity.entity_description.native_max_value = (
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'native_max_value'
```

Thoughts?

Reply:
You can set _attr_native_max_value in the entity before adding it to the entities list

Reply:
I am really sorry this seems to be more difficult than I thought it would. I have been looking at this for a while and cannot get anything working. 

Here is what I have tried recently. Am I missing something? 

(Sorry, I havent written a ton of python in the past) 

```
        for core_climate in bed.foundation.core_climates:
            entity = SleepIQNumberEntity(
                data.data_coordinator,
                bed,
                core_climate,
                NUMBER_DESCRIPTIONS[CORE_CLIMATE_TIMER],
            )

            entity._attr_native_max_value = core_climate.max_core_climate_time

            entities.append(entity)
```

Results in: 

```
2024-01-28 22:59:17.848 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [281472299496512] 
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/number/__init__.py", line 111, in async_set_value
    await entity.async_set_native_value(native_value)
  File "/workspaces/core/homeassistant/components/number/__init__.py", line 400, in async_set_native_value
    await self.hass.async_add_executor_job(self.set_native_value, value)
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/number/__init__.py", line 396, in set_native_value
    raise NotImplementedError()
NotImplementedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/websocket_api/commands.py", line 240, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/workspaces/core/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/service.py", line 887, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/service.py", line 957, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/number/__init__.py", line 113, in async_set_value
    await entity.async_set_value(value)
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/number/__init__.py", line 405, in set_value
    raise NotImplementedError()
NotImplementedError
```